src/components/MapView/DateSelector/utils.ts

```diff
@@ -15,6 +15,48 @@ export const months = [
   'Dec',
 ];
 
-export function findAvailableDayInMonth(date: Date, availableDates: Date[]) {
-  return availableDates.find(dateIt => moment(dateIt).isSame(date, 'month'));
+/**
+ * Return the start and end date of a month (utc format)
+ * @param month
+ * @param year
+ * @return { startDate, endDate }
+ */
+export function getMonthStartAndEnd(month: number, year: number) {
+  const monthDate = moment({ year, month });
+
+  const startDate = monthDate.startOf('month').valueOf();
+  const endDate = monthDate.endOf('month').valueOf();
+
+  return { startDate, endDate };
+}
+
+/**
+ * Return the list of available dates the month
+ * @param month
+ * @param year
+ * @param availableDates
+ * @return a list of available dates the month
+ */
+export function findAvailableDayInMonth(
+  month: number,
+  year: number,
+  availableDates: Date[],
+) {
+  const date = new Date(year, month);
+  return availableDates.filter(dateIt => moment(dateIt).isSame(date, 'month'));
+}
+
+/**
+ * Return true if there is an available date in the month
+ * @param month
+ * @param year
+ * @param availableDates
+ * @return
+ */
+export function isAvailableMonth(
+  month: number,
+  year: number,
+  availableDates: Date[],
+) {
+  return findAvailableDayInMonth(month, year, availableDates).length;
```

Comment:
Let's convert this to a `boolean` - especially using an inferred return type, Typescript is going to "think" of this function as returning an integer, not a boolean